### ISSUE TITLE
ucontroller example broken

### ISSUE THREAD

**Issue by jeffwitz** [LaboratoireMecaniqueLille/crappy#120]
Hi,

We try to use the example of ucontroller communication of crappy, but it appears that it doesn't work anymore.

Nothing is read by the crappy program , so nothing is displayed inside the dash.

**Comment by WeisLeDocto** [comment]
Please be more specific. As per the documentation of the [`ucontroller.py` example](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/examples/blocks/ucontroller/ucontroller.py):

> Before starting this script, you first have to install MicroPython on your microcontroller and upload the microcontroller.py script as the main.py file.

Did you follow these steps ? 
Also:

> It was written for and tested on Adafruit's Huzzah ESP32 board, but should work on other microcontrollers or boards with minor adjustments.

What microcontroller are you using ? Which version of MicroPython ? 
Finally:

> The UController Block can drive a microcontroller with MicroPython installed (...) It can also drive a microcontroller running the Arduino microcontroller.ino script, that can be found in Crappy's tool.

This holds true for the `UController` Block in the general case, but not for the [`ucontroller.py` example](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/examples/blocks/ucontroller/ucontroller.py) example specifically. This example was not tested using the Arduino code, there is therefore no guarantee that it works as is with the [`microcontroller.ino`](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/src/crappy/tool/microcontroller.ino) script.

**Comment by WeisLeDocto** [comment]
Just as a quick update, I tested the [`ucontroller.py` example](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/examples/blocks/ucontroller/ucontroller.py) on a 2ac7z-esp32 microcontroller (Adafruit Huzzah32 Feather), along with the [`microcontroller.py` script](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/examples/blocks/ucontroller/microcontroller.py), and everything worked as expected.

Steps that I took:

- Flash MicroPython on the board ([`esptool.py`](https://github.com/espressif/esptool) and [`esp32-20210902-v1.17.bin`](https://micropython.org/download/ESP32_GENERIC/) or later available online)
  ```
  esptool.py --port /dev/ttyUSB0 erase_flash
  esptool.py -b 115200 --port /dev/ttyUSB0 write_flash 0x1000 esp32-20210902-v1.17.bin
  ```
- Copy the `microcontroller.py` file ([`pyboard.py`](https://github.com/micropython/micropython/blob/master/tools/pyboard.py) available online)
  ```
  pyboard.py -d /dev/ttyUSB0 -f cp microncontroller.py : main.py
  ```
- Reset the ESP32 using the reset button
- Start the `ucontroller.py` example

**Comment by WeisLeDocto** [comment]
New update, I also tested the [`microcontroller.ino` script](https://github.com/LaboratoireMecaniqueLille/crappy/blob/e069fd4c6cf7209c56833290a24f3f4235c9684c/src/crappy/tool/microcontroller.ino) on the same device as in my previous message, and obtained successful results. 

I'm now closing this issue, as I can't reproduce the (very imprecisely) described bug, and as everything seems to be functional on Crappy's side. Keep in mind that the `microcontroller.py` and `microcontroller.ino` scripts may need to be adapted to each new chip + board combination, and might as well not work at all in some cases. As such adaptations are specific to each user, they fall outside the scope of the support I can offer.

Steps that I took:

- Compile and flash the `microcontroller.ino` script using the Arduino IDE (1.8.20 Hourly Build 2021/12/20 07:25, Port Adafruit ESP32 Feather)
- Reset the ESP32 using the reset button
- Start the ucontroller.py example
